pkg/client/testing/core/fixture.go

```diff
@@ -66,6 +66,9 @@ func ObjectReaction(tracker ObjectTracker, mapper meta.RESTMapper) ReactionFunc
 	return func(action Action) (bool, runtime.Object, error) {
 		ns := action.GetNamespace()
 		gvr := action.GetResource()
+		if action.GetSubresource() == "scale" {
```

Comment:
I struggled with a clean way. If I treat all subresources the same, things go poorly. It seems subresources are all treated equal throughout which is why I ended up with the special case for scale. I am open to suggestions.